pallets/pallet-bank/src/benchmarking.rs

```diff
@@ -0,0 +1,175 @@
+//! Benchmarking setup for pallet-template
+#![cfg(feature = "runtime-benchmarks")]
+use super::*;
+
+use frame_benchmarking::v2::*;
+use frame_support::assert_ok;
+use frame_system::RawOrigin;
+
+#[benchmarks]
+mod benchmarks {
+
+	use super::*;
+	struct MockUsers<AccountId> {
+		pub manager: AccountId,
+		pub auditor: AccountId,
+		pub customer_1: AccountId,
+		pub customer_2: AccountId,
+	}
+
+	fn setup<T: Config>() -> MockUsers<T::AccountId> {
+		let manager: T::AccountId = whitelisted_caller();
+		let auditor: T::AccountId = account("auditor", 0u32, 0u32);
+		let customer_1: T::AccountId = account("customer_1", 0u32, 0u32);
+		let customer_2: T::AccountId = account("customer_2", 0u32, 0u32);
+		assert_ok!(T::RoleManager::register_role(&manager, Role::Manager));
+		assert_ok!(T::RoleManager::register_role(&auditor, Role::Auditor));
+		assert_ok!(T::RoleManager::register_role(&customer_1, Role::Customer));
+		assert_ok!(T::RoleManager::register_role(&customer_2, Role::Customer));
+		Accounts::<T>::insert(
+			&customer_1,
+			AccountData {
+				free: 1_000_000u128.into(),
+				reserved: 1_000_000u128.into(),
+				locked: vec![],
+			},
+		);
+		Accounts::<T>::insert(
+			&customer_2,
+			AccountData {
+				free: 1_000_000u128.into(),
+				reserved: 1_000_000u128.into(),
+				locked: vec![LockedFund {
+					id: 1u64.into(),
+					amount: 1_000u128.into(),
+					reason: LockReason::Auditor,
+				}],
+			},
+		);
+
+		MockUsers { manager, auditor, customer_1, customer_2 }
+	}
+
+	#[benchmark]
+	fn deposit() {
+		let accounts = setup::<T>();
+		let initial_balance = Accounts::<T>::get(&accounts.customer_1).free;
+		let amount = 1_000u128.into();
+
+		#[extrinsic_call]
+		deposit(RawOrigin::Signed(accounts.manager), accounts.customer_1.clone(), amount);
+
+		// Verify
+		assert_eq!(Accounts::<T>::get(accounts.customer_1).free, initial_balance + amount);
+	}
+
+	#[benchmark]
+	fn withdraw() {
+		let accounts = setup::<T>();
+		let amount = 1_000u128.into();
+		let initial_balance = Accounts::<T>::get(&accounts.customer_1).free;
+		#[extrinsic_call]
+		withdraw(RawOrigin::Signed(accounts.manager), accounts.customer_1.clone(), amount);
+
+		// Verify
+		assert_eq!(Accounts::<T>::get(&accounts.customer_1).free, initial_balance - amount);
+	}
+
+	#[benchmark]
+	fn transfer() {
+		let accounts = setup::<T>();
+		let amount = 1_000u128.into();
+		let initial_balance = Accounts::<T>::get(&accounts.customer_1).free;
```

Comment:
more correct to check initial balance of both customer 1 and 2, instead of using the same "initial_balance" for both accounts